src/components/vm/overview/firmware.jsx

```diff
@@ -52,10 +52,23 @@ class FirmwareModal extends React.Component {
         this.setState({ dialogError: text, dialogErrorDetail: detail });
     }
 
-    save() {
+    async save() {
         const Dialogs = this.context;
-        domainSetOSFirmware({ connectionName: this.props.connectionName, objPath: this.props.vmId, loaderType: stateToXml(this.state.firmware) })
-                .then(Dialogs.close, exc => this.dialogErrorSet(_("Failed to change firmware"), exc.message));
+        const vm = this.props.vm;
+        try {
+            await domainSetOSFirmware({
+                connectionName: vm.connectionName,
+                objPath: vm.id,
+                loaderType: stateToXml(this.state.firmware)
+            });
+
+            if (this.state.firmware == 'efi' && !vm.hasTPM && vm.capabilities.supportsTPM)
+                await domainAddTPM({ connectionName: vm.connectionName, vmName: vm.name });
+
+            Dialogs.close();
+        } catch (exc) {
+            this.dialogErrorSet(_("Failed to change firmware"), exc.message);
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1448-20240219-070059-397d4f2a-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)